packages/m2m-gateway/src/services/eserviceService.ts

```diff
@@ -170,7 +170,57 @@ export function eserviceServiceBuilder(
       totalCount: allFlatKindAttributes.length,
     };
   }
+  async function createEServiceDescriptorAttributesGroup(
+    eserviceId: EServiceId,
+    descriptorId: DescriptorId,
+    attributeIds: string[],
+    attributeKind: keyof catalogApi.Attributes,
+    { headers }: WithLogger<M2MGatewayAppContext>
+  ): Promise<{
+    groupIndex: number;
+    attributes: attributeRegistryApi.Attributes;
+  }> {
+    const eservice = await retrieveEServiceById(headers, eserviceId);
+    const descriptor = retrieveEServiceDescriptorById(eservice, descriptorId);
+    // Get the new group index (will be the length of current groups)
+    const newGroupIndex = descriptor.attributes[attributeKind].length;
+    const newAttributeGroups = [
+      ...descriptor.attributes[attributeKind],
+      attributeIds.map((id) => ({
+        id,
+        explicitAttributeVerification: false,
+      })),
+    ];
+    const newAttributes = {
+      ...descriptor.attributes,
+      [attributeKind]: newAttributeGroups,
+    };
+    // Update the descriptor with the new attributes
+    const response = await clients.catalogProcessClient.updateDescriptorAttributes(
```

Comment:
The creation of a new group should be draft-only, while this is the method to update attributes of a published descriptor. See what @Carminepo2 did in the DELETE https://github.com/pagopa/interop-be-monorepo/pull/2504 - you should reconstruct the attribute groups of the kind and call the PATCH for the draft